src/cljs/rems/table.cljs

```diff
@@ -94,20 +140,38 @@
                               (s/optional-key :sort-column) s/Keyword
                               (s/optional-key :sort-order) (s/enum :asc :desc)
                               (s/optional-key :set-sorting) Applicable}
-   (s/optional-key :filtering) {(s/optional-key :filters) {s/Keyword s/Str}
+   (s/optional-key :filtering) {(s/optional-key :filters) s/Str
                                 (s/optional-key :show-filters) s/Bool
                                 (s/optional-key :set-filtering) Applicable}
    :id-function Applicable
    :items [s/Any]
    (s/optional-key :class) s/Str})
 
+(defn- filter-view [filtering]
+  (let [{:keys [filters set-filtering]} filtering]
+    [:div
+      [:input
+       {:type "text"
+        :name "table-search"
+        :value filters
+        :placeholder ""
+        :on-input (fn [event]
+                    (set-filtering (assoc filtering :filters (-> event .-target .-value))))}]
```

Comment:
Maybe this would be a good place for preparing the filters with a helper fn that parses the input value? You could also test that separately.

Reply:
Disagree, if we save a list instead of string to the db, we would have to then generate the string again (the input is a controlled component).